### PR TITLE
[MOL-15934][AT] Add support for an isReady state to handle situation when download is still being generated or not available to the user.

### PR DESCRIPTION
**Pull request by thyAshley**
**Changes**

**FileDownload Component**
Add a `isReady` prop to fileItemDownload which is defaulted to `true`
Add a check in handleDownload that prevents user from downloading the file when either isLoading  is true or when isReady is false

- delete branch

**Changelog entry**
Add support for an `isReady` state to handle situation when download is still being generated or not available to the user.

**Additional information**

- You may refer to this [MOL-15934](https://jira.ship.gov.sg/browse/MOL-15934)

**demo**

[Demo.webm](https://github.com/user-attachments/assets/0dd4fa4b-58fc-4422-bfc4-09b38a36dd2e)

### CODE REVIEW

File: src/file-download/types.ts [hunk 1]
```diff
@@ -14,6 +14,8 @@ export interface FileItemDownloadProps {
     thumbnailImageDataUrl?: string | undefined;
     /** Indicates if text should be truncated */
     truncateText?: boolean | undefined;
+    /** Indicate if file is ready for download, defaults to true */
+    isReady?: boolean;
```
Comment (qroll) [inline src/file-download/types.ts:18]:
```suggestion
    ready?: boolean | undefined;
```

File: stories/file-download/file-download.stories.tsx [hunk 2]
```diff
@@ -52,7 +52,7 @@ export const Default: StoryObj<Component> = {
 
         return (
             <FileDownload
-                fileItems={fileItems}
+                fileItems={[fileItems[2]]}
```
Comment (qroll) [inline stories/file-download/file-download.stories.tsx:55]: is this change intended? as it's for a separate story

File: stories/file-download/file-download.stories.tsx
Comment (qroll) [inline stories/file-download/file-download.stories.tsx:1]: the story should be included in the main `file-download.mdx` as well, with a short explanation describing the use case

File: src/file-download/types.ts
Comment (qroll) [inline src/file-download/types.ts:1]: update `stories/file-download/props-table.tsx` with the new prop as well

**Review by qroll — CHANGES_REQUESTED** [review]

File: stories/file-download/file-download.stories.tsx (hunk 2 above)
Comment (thyAshley) [reply stories/file-download/file-download.stories.tsx:55]: oh shit, copy pasta lol

**Review by qroll — APPROVED** [review]
